src/test/java/calculator/StringCalculatorTest.kt

```diff
@@ -0,0 +1,61 @@
+package calculator
+
+import org.assertj.core.api.Assertions.assertThat
+import org.assertj.core.api.AssertionsForClassTypes.assertThatExceptionOfType
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.NullAndEmptySource
+import org.junit.jupiter.params.provider.ValueSource
+
+class StringCalculatorTest {
+    private lateinit var calculator: StringCalculator
+
+    @BeforeEach
+    fun setUp() {
+        calculator = StringCalculator()
+    }
+
+    @DisplayName("빈 문자열 또는 null 값을 입력할 경우 0을 반환해야 한다.")
+    @ParameterizedTest
+    @NullAndEmptySource
+    fun test1(text: String?) {
+        assertThat(calculator.add(text)).isZero()
+    }
+
+    @DisplayName("숫자 하나를 문자열로 입력할 경우 해당 숫자를 반환한다.")
+    @ParameterizedTest
+    @ValueSource(strings = ["1", "11", "0", "99"])
+    fun test2(text: String) {
+        assertThat(calculator.add(text)).isSameAs(text.toInt())
+    }
+
+    @DisplayName("숫자 두개를 쉼표(,) 구분자로 입력할 경우 두 숫자의 합을 반환한다.")
+    @ParameterizedTest
+    @ValueSource(strings = ["1,2"])
+    fun test3(text: String) {
+        assertThat(calculator.add(text)).isSameAs(3)
+    }
+
+    @DisplayName("구분자를 쉼표(,) 이외에 콜론(:)을 사용할 수 있다.")
+    @ParameterizedTest
+    @ValueSource(strings = ["1,2:3"])
+    fun test4(text: String) {
+        assertThat(calculator.add(text)).isSameAs(6)
+    }
+
+    @DisplayName("//와 \\n 문자 사이에 커스텀 구분자를 지정할 수 있다.")
+    @ParameterizedTest
+    @ValueSource(strings = ["//;\n1;2;3"])
```

Comment:
만약 커스텀 구분자가 마침표라면 테스트 결과가 어떻게 될까요? 🤔
```suggestion
    @ValueSource(strings = ["//.\n1.2.3"])
```

Reply:
이스케이프 문자를 생각못했네요. 해당 부분 [187ed2b](https://github.com/next-step/ddd-legacy/pull/586/commits/187ed2be56b4c135c7ba9e4a87359d3117a9b5b1) 에서 수정 완료 하였습니다.

Reply:
항상 모든 경우의 수를 신경써야하는게 개발자의 운명인 것 같아요 😅 
한가지 작은 팁을 드리자면 다음 코드처럼 Pattern.quote()를 사용해 정규식표현문자를 리터럴로 인식하게 할 수 있습니다.
`Pattern.quote(matcher.group(DELIMITER_GROUP_INDEX))`

Reply:
좋은 팁 감사합니다! 다음 부터는 주어진 케이스뿐만 아니라, 모든 경우의 수를 생각하도록 노력하겠습니다.

[5428cbb](https://github.com/next-step/ddd-legacy/pull/586/commits/5428cbb4b5ed08d5b0714204a0b10f4adf2ef5e6) 해당 커밋에서 수정 완료하였습니다.